server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1729,6 +1754,21 @@ public void snapshotShard(Store store, MapperService mapperService, SnapshotId s
         }
     }
 
+    @Nullable
+    private static List<BlobStoreIndexShardSnapshot.FileInfo> findMatchingShardSnapshot(long globalCheckpoint, long maxSequenceNum,
+                                                                                        String historyUUID,
+                                                                                        BlobStoreIndexShardSnapshots snapshots) {
+        if (maxSequenceNum == SequenceNumbers.UNASSIGNED_SEQ_NO || globalCheckpoint != maxSequenceNum) {
```

Comment:
I think this logic should live in the caller where we determine whether we want to do any kind of lookup by this special number (need a name for it) that is equal to global checkpoint, local checkpoint, and max sequence number.

Reply:
Right, this makes everything so much simpler => moved this to `SnapshotShardsService` now instead of messing with the explicit global checkpoint in the repository.